### ISSUE TITLE
Getting this job was abandoned error message while building on macOS-13 vs. macOS-12

### ISSUE THREAD

**Issue by antonioalwan** [actions/runner-images#7956]
### Description

I am getting intermittently every time i ran the pipeline a `job was abandoned` error from different agents:
`This job was abandoned. We have detected that logs from the agent may have not finished uploading. We have included our in-memory record of all log lines uploaded before we lost contact with the agent:
Starting: Run automation tests`
This same error doesn't happen when using macOS-12. 
Error:
<img width="1952" alt="image" src="https://github.com/actions/runner-images/assets/11672290/d02e814c-190e-4f6e-bec8-982d761c68da">

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [X] macOS 13
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1138769&view=results

### Is it regression?

No

### Expected behavior

To run successfully without `job was abandoned` error

### Actual behavior

As you see in the image, the task is not even complete and is around 45 mins through.. 
This issue has been preventing us from switching our automation tests to use macOS-13. This pipeline run multiple stages in parallel out of which every time it is running a consistent random-stage error message is displayed and the error is causing the stage to fail. This never happens with macOS-12. Pipeline view of stages here:
<img width="2086" alt="image" src="https://github.com/actions/runner-images/assets/11672290/d7f7429c-59ff-4de4-a3cd-f1d6d58d6296">
We'd like to have a resolution or suggestion on how to fix this issue?

### Repro steps

Run the listed pipeline

**Comment by mikhailkoliada** [comment]
It is a duplicate of #7754
